vendor/bat-native-ads/src/bat/ads/internal/ads/inline_content_ads/inline_content_ad.cc

```diff
@@ -73,25 +73,31 @@ void InlineContentAd::FireEvent(
     const std::string& creative_instance_id,
     const mojom::InlineContentAdEventType event_type) {
   database::table::AdEvents database_table;
-  database_table.GetAll([=](const bool success, const AdEventList& ad_events) {
-    if (!success) {
-      BLOG(1, "Inline content ad: Failed to get ad events");
-      NotifyInlineContentAdEventFailed(uuid, creative_instance_id, event_type);
-      return;
-    }
+  database_table.GetForType(
+      mojom::AdType::kInlineContentAd,
```

Comment:
We now only get a list of relevant ad types, and then calculate the eligible ads for that type. Previously we got all ad events and filtered other ad types within the frequency caps